public/app/AppWrapper.tsx

```diff
@@ -109,33 +125,21 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
                 actions={[]}
                 options={{ enableHistory: true, callbacks: { onSelectAction: commandPaletteActionSelected } }}
               >
-                <Router history={locationService.getHistory()}>
-                  <LocationServiceProvider service={locationService}>
-                    <CompatRouter>
-                      <ModalsContextProvider>
-                        <GlobalStyles />
-                        <div className="grafana-app">
-                          <AppChrome>
-                            <AngularRoot />
-                            <AppNotificationList />
-                            <Stack gap={0} grow={1} direction="column">
-                              {pageBanners.map((Banner, index) => (
-                                <Banner key={index.toString()} />
-                              ))}
-                              {ready && this.renderRoutes()}
-                            </Stack>
-                            {bodyRenderHooks.map((Hook, index) => (
-                              <Hook key={index.toString()} />
-                            ))}
-                          </AppChrome>
-                        </div>
-                        <LiveConnectionWarning />
-                        <ModalRoot />
-                        <PortalContainer />
-                      </ModalsContextProvider>
-                    </CompatRouter>
-                  </LocationServiceProvider>
-                </Router>
+                <ModalsContextProvider>
+                  <GlobalStyles />
+                  <SidecarContext.Provider value={sidecarService}>
+                    <div className="grafana-app">
+                      {config.featureToggles.appSidecar ? (
+                        <ExperimentalSplitPaneTree routes={ready && this.renderRoutes()} />
+                      ) : (
+                        <RouterTree routes={ready && this.renderRoutes()} />
+                      )}
+                      <LiveConnectionWarning />
+                      <ModalRoot />
+                      <PortalContainer />
```

Comment:
Don't think these can be outside the router

Reply:
I don't think they directly need router context. I assume if things are portaled over the component gets the context from where it is defined so if there is a link in the context of the modal or some dropdown it will carry over it's own context but will check to be sure.

Reply:
So yeah the Portal is fine so dropdowns etc get the correct context. With modals this is but more problematic I assume because we use the event bus to pass components around. This means the ModalProvider and ModalRoot need to be inside a Router/Location context. We could have 2 of them, don't think that would be a big deal but the issue is that the eventBus is still a singleton so the events would have been subscribed 2 times. At this point I would leave it as it is and try to deal with it later. With the current code the issue is only if a modal inside the second app tries to do something with routing, it would do it on a global level, ie changing the global URL.

Reply:
But leaving as it would introduce modal bugs, have had issues with links inside modals not working properly due to not being inside router context

Reply:
Yeah I changed it so that modal context is inside a router here https://github.com/grafana/grafana/pull/91648/commits/9ca3a7a856f5c82e1ca84b9d347b43d57bde7bb9#diff-b1d459ae81cb312ba78da28cfe277013ba08a6787c0f5dcc42ac142f06dadd6cR154. It just means that if modal is used inside a sidecar app it will act upon a main router for navigation for example. Will add it to stuff to fix but again not sure I want to add it to this particular PR as that would probably mean changing some more singletons.